cloudinit/sources/DataSourceVMware.py

```diff
@@ -182,6 +212,12 @@ def _get_data(self):
 
         LOG.info("using data access method %s", self._get_subplatform())
 
+        # The default list of configured update events should be the same as
+        # the list of supported update events.
+        self.default_update_events = copy.deepcopy(
```

Comment:
There's no reason to include `BOOT_LEGACY` as a default anymore. It only exists for backwards compatibility. Cloud-init has 4 boot stages, and BOOT_LEGACY means retrieve the IMDS metadata in both of the first two stages. This is unnecessary because if it is retrieved in the 1st stage, it then gets pickled and used in the 2nd stage. If it fails in the 1st stage, we try again in the 2nd stage.